# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/JitActivation.cpp
# Commit: caa8552f4f52
# Full Hash: caa8552f4f5298399e9e9193a623bc3007652df5
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-06-18 09:41:05
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Refactor wasm::Frame to reduce number of gnarly casts. r=lth,wingo
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D79072
# ==============================================================================

diff -r 43921ad138e3 -r caa8552f4f52 js/src/vm/JitActivation.cpp
--- a/js/src/vm/JitActivation.cpp	Mon Jun 08 20:56:09 2020 +0000
+++ b/js/src/vm/JitActivation.cpp	Thu Jun 18 05:33:30 2020 +0000
@@ -231,9 +231,9 @@
   MOZ_ASSERT(unwound == (trap == wasm::Trap::IndirectCallBadSig));
 
   void* pc = unwindState.pc;
-  wasm::Frame* fp = unwindState.fp;
+  const wasm::Frame* fp = wasm::Frame::fromUntaggedWasmExitFP(unwindState.fp);
 
-  const wasm::Code& code = fp->tls->instance->code();
+  const wasm::Code& code = fp->instance()->code();
   MOZ_RELEASE_ASSERT(&code == wasm::LookupCode(pc));
 
   // If the frame was unwound, the bytecodeOffset must be recovered from the